code/modules/mob/living/basic/farm_animals/bee/_bee.dm

```diff
@@ -0,0 +1,338 @@
+		col = beegent.color
+
+	icon_state = "[icon_base]_base"
+	add_overlay("[icon_base]_base")
+
+	var/static/mutable_appearance/greyscale_overlay
+	greyscale_overlay = greyscale_overlay || mutable_appearance('icons/mob/simple/bees.dmi')
+	greyscale_overlay.icon_state = "[icon_base]_grey"
+	greyscale_overlay.color = col
+	add_overlay(greyscale_overlay)
+
+	add_overlay("[icon_base]_wings")
+
+/mob/living/basic/bee/proc/pollinate(obj/machinery/hydroponics/hydro)
+
+	hydro.recent_bee_visit = TRUE
+	addtimer(VARSET_CALLBACK(hydro, recent_bee_visit, FALSE), BEE_TRAY_RECENT_VISIT)
+
+	var/growth = health //Health also means how many bees are in the swarm, roughly.
+	//better healthier plants!
+	hydro.adjust_plant_health(growth*0.5)
+	if(prob(BEE_POLLINATE_PEST_CHANCE))
+		hydro.adjust_pestlevel(-10)
+	if(prob(BEE_POLLINATE_YIELD_CHANCE))
+		hydro.myseed.adjust_yield(1)
+		hydro.yieldmod = 2
+	if(prob(BEE_POLLINATE_POTENCY_CHANCE))
+		hydro.myseed.adjust_potency(1)
+
+	if(beehome)
+		beehome.bee_resources = min(beehome.bee_resources + growth, 100)
+
+/mob/living/basic/bee/proc/assign_reagent(datum/reagent/toxin)
+	if(istype(toxin))
+		var/static/list/injection_range = list(1, 5)
+		if(beegent) //clear the old since this one is going to have some new value
+			RemoveElement(/datum/element/venomous, beegent.type, injection_range)
+		beegent = toxin
+		name = "[initial(name)] ([toxin.name])"
+		real_name = name
+		AddElement(/datum/element/venomous, beegent.type, injection_range)
+		generate_bee_visuals()
+
+/mob/living/basic/bee/queen
+	name = "queen bee"
+	desc = "She's the queen of bees, BZZ BZZ!"
+	icon_base = "queen"
+	isqueen = TRUE
+	ai_controller = /datum/ai_controller/basic_controller/queen_bee
+
+/mob/living/basic/bee/queen/will_escape_storage()
+	return FALSE
+
+/mob/living/basic/bee/toxin
+
+/mob/living/basic/bee/toxin/Initialize(mapload)
+	. = ..()
+	var/datum/reagent/toxin = pick(typesof(/datum/reagent/toxin))
+	assign_reagent(GLOB.chemical_reagents_list[toxin])
+
+/mob/living/basic/bee/short
+	desc = "These bees seem unstable and won't survive for long."
+
+/mob/living/basic/bee/short/Initialize(mapload, timetolive=50 SECONDS)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(death)), timetolive)
+
+/obj/item/queen_bee
+	name = "queen bee"
+	desc = "She's the queen of bees, BZZ BZZ!"
+	icon_state = "queen_item"
+	inhand_icon_state = ""
+	icon = 'icons/mob/simple/bees.dmi'
+	/// The actual mob that our bee item corresponds to
+	var/mob/living/basic/bee/queen/queen
+
+/obj/item/queen_bee/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_QUEEN_BEE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+/obj/item/queen_bee/Destroy()
+	QDEL_NULL(queen)
+	return ..()
+
+/obj/item/queen_bee/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == queen)
+		queen = null
+		// the bee should not exist without a bee.
+		if(!QDELETED(src))
+			qdel(src)
+
+/obj/item/queen_bee/attackby(obj/item/syringe, mob/user, params)
```

Comment:
again, since we override all of `attackby()`, i feel like signals would be optimal?